content/usgs-packages/sbtools_Modify.Rmd

```diff
@@ -0,0 +1,302 @@
+```
+
+Next, add three items at once. Nest the first two items under the new folder, and the last one as a top-level folder. Note that this currently does not do the expected behavior - it just puts everything under the new folder. This is a [known issue](https://github.com/USGS-R/sbtools/issues/242) in sbtools.
+
+```{r add-multiple-items}
+add_mult <- items_create(parent_id = list(new_folder, new_folder, user_id()),
+                         title = c("item 1", "item 2", "top-level item"))
+
+# verify that all three were successfully added
+sapply(add_mult, function(item) is.sbitem(item_get(item)))
+
+# verify that they were put in the correct places
+c("usgs-r-pkgs-test", "usgs-r-pkgs-test", item_get(user_id())$title) ==
+  sapply(add_mult, function(item) item_get_parent(item)$title)
+```
+
+If you want to confirm with your own eyes, navigate to your user account. You should see two items: "top-level item" and the folder "usgs-r-pkgs-test" with 3 child items named "single item", "item 1", and "item 2". **AGAIN, "top-level item" will not actually be top-level until [this issue](https://github.com/USGS-R/sbtools/issues/242) has been fixed, so the third item will return FALSE for the second verification step.**
+
+## Uploading your files
+
+You can upload files directly from R to an existing or new ScienceBase item. All you need to do is select the appropriate function and get the file names. There are a few helper functions from base R that might be useful when working with files and filepaths: `list.dirs`, `list.files`, `file.path`, `dir.exists`, `file.exists`, `system.file`. Some may be shown in these examples, but explore them in more detail if they are unfamiliar.
+
+This section will give examples for the following functions: 
+
+* `item_upload_create`
+* `item_append_files`
+
+To upload files to a new ScienceBase item, use the function `item_upload_create`. This function creates a new item, and uploads new files simultaneously. Try it out by uploading the first two files available in the `sbtools` package to a new item.
+
+```{r sbtools-upload}
+dir <- system.file("examples", package="sbtools")
+sbtools_files <- list.files(dir)[1:2]
+sbtools_files
+sbtools_filepaths <- file.path(dir, sbtools_files)
+test_item <- item_upload_create(parent_id = user_id(),
+                                files = sbtools_filepaths)
+test_item
+```
+
+Currently, the title of the new item defaults to the first file that is uploaded, and there is no argument to override this behavior. This is a known `sbtools` [issue](https://github.com/USGS-R/sbtools/issues/49). For now, you can change the title using the function `item_update`, which will be discusssed later in this lesson. Another interesting behavior is that uploading multiple files at once sometimes does not work for complex files. This is a [known issue](https://github.com/USGS-R/sbtools/issues/39). In the meantime, always verify that your files were actually uploaded.
```

Comment:
:+1: agree! Tim and I chatted over email when I had that crazy "deleted my home item" issue. Sounds like ScienceBase 3.0 comes out in a few months?